### ISSUE TITLE
h2o SSL - chrome reporting insecure content when there is none ?

### ISSUE THREAD

**Issue by centminmod** [h2o/h2o#231]
I am using H2O 1.1.1 with SSL config outlined at https://github.com/h2o/h2o/issues/230

**Problem**
When I view the H2O SSL served site on 8081 pages and even individual images loaded in Chrome are reported by Chrome as having insecure content served. But how is that possible even when loading an individual image i.e. cmlogo.png over https:// ? Chrome / HTTP/2 issue or H2O issue ?

* Chrome Version 41.0.2272.89 m
* Windows 8 64bit

![](http://centminmod.com/h2o/screenshots/ssl/h2o_insecure_content_00.png)

check https SSL on port 8081 for H20 server using [cipherscan](https://github.com/jvehent/cipherscan)

    ./cipherscan XXX.centminmod.com:8081        
    ..............
    Target: XXX.centminmod.com:8081
    
    prio  ciphersuite                  protocols              pfs_keysize
    1     ECDHE-RSA-AES256-GCM-SHA384  TLSv1.2                ECDH,P-256,256bits
    2     ECDHE-RSA-AES256-SHA384      TLSv1.2                ECDH,P-256,256bits
    3     ECDHE-RSA-AES256-SHA         TLSv1,TLSv1.1,TLSv1.2  ECDH,P-256,256bits
    4     AES256-GCM-SHA384            TLSv1.2
    5     AES256-SHA256                TLSv1.2
    6     AES256-SHA                   TLSv1,TLSv1.1,TLSv1.2
    7     ECDHE-RSA-AES128-GCM-SHA256  TLSv1.2                ECDH,P-256,256bits
    8     ECDHE-RSA-AES128-SHA256      TLSv1.2                ECDH,P-256,256bits
    9     ECDHE-RSA-AES128-SHA         TLSv1,TLSv1.1,TLSv1.2  ECDH,P-256,256bits
    10    AES128-GCM-SHA256            TLSv1.2
    11    AES128-SHA256                TLSv1.2
    12    AES128-SHA                   TLSv1,TLSv1.1,TLSv1.2
    13    DES-CBC3-SHA                 TLSv1,TLSv1.1,TLSv1.2
    
    Certificate: trusted, 2048 bit, sha256WithRSAEncryption signature
    TLS ticket lifetime hint: 300
    OCSP stapling: supported
    Client side cipher ordering

**Comment by kazuho** [comment]
@centminmod The issue seems to be either in Chrome or your configuration.

I would suggest: 1) check that all resources are accessed via HTTPS by using the Developer Tools of Chrome 2) use Firefox to see if the problem is in Chrome or not.

**Comment by centminmod** [comment]
yeah the above image bottom half is Chrome developer tools open for cmlogo.png image - just the standalone image invoked the Chrome insecure content/resource warning

**Comment by centminmod** [comment]
checking h2o access log, only cmlogo.png is being requested it seems

    - - [16/Mar/2015:00:25:22 +0000] "GET / HTTP/2" 200 3801 "-" "Mozilla/5.0 (Windows NT 6.2; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.89 Safari/537.36"
    - - [16/Mar/2015:02:11:30 +0000] "GET /cmlogo.png HTTP/2" 200 8934 "-" "Mozilla/5.0 (Windows NT 6.2; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.89 Safari/537.36"

**Comment by kazuho** [comment]
@centminmod does other web browsers (e.g. Firefox as we are using HTTP/2) report the same issue?

**Comment by centminmod** [comment]
don't use Firefox, but checked with Opera 28 and no such reported insecure content, I get full green padlock, but Opera 28 detects the H2O served port 8081 https as SPDY even after I used chrome flags to enable SPDY/4 and restart Opera 28 browser

**Comment by kazuho** [comment]
@centminmod 
> guess it might be Chrome issue ?

It might be, and it might be specific to Windows.

FYI running H2O 1.1.1 and Chrome 41 on OSX I could not reproduce the issue.  Chrome did not report any errors as can bee seen on http://i.gyazo.com/cb4d8a37344ca66f8d368d82678ab4cf.png

**Comment by centminmod** [comment]
thanks @kazuho will check in Chrome on other devices i.e. mobile/tablet etc as well

**Comment by lkwg82** [comment]
still an issue?
